sockeye/training.py

```diff
@@ -967,41 +967,38 @@ def _load_training_state(self, train_iter: data_io.BaseParallelSampleIter):
 
 class TensorboardLogger:
     """
-    Thin wrapper for MXBoard API to log scalar values.
+    Thin wrapper for MXBoard API to log training events.
     """
 
     def __init__(self, logdir: str) -> None:
         self.logdir = logdir
-
         try:
             import mxboard
             logger.info("Logging training events for Tensorboard at '%s'", self.logdir)
+            if os.path.exists(self.logdir):
+                logger.info("Deleting existing Tensorboard log directory '%s'", self.logdir)
+                shutil.rmtree(self.logdir)
+            self.sw = mxboard.SummaryWriter(logdir=self.logdir)
         except ImportError:
             logger.info("mxboard not found. Consider 'pip install mxboard' to log events to Tensorboard.")
-
-        if os.path.exists(self.logdir):
-            logger.info("Deleting existing Tensorboard log directory '%s'", self.logdir)
-            shutil.rmtree(self.logdir)
+            self.sw = None
 
     def log_metrics(self, metrics: Dict[str, Union[float, int, mx.nd.NDArray]], checkpoint: int):
-        try:
-            from mxboard import SummaryWriter
-            with SummaryWriter(logdir=self.logdir) as sw:
-                for name, value in metrics.items():
-                    if isinstance(value, mx.nd.NDArray):
-                        sw.add_histogram(tag=name, values=value, bins=100, global_step=checkpoint)
-                    else:
-                        sw.add_scalar(tag=name, value=value, global_step=checkpoint)
-        except ImportError:
+        if self.sw is None:
             return
 
+        for name, value in metrics.items():
+            if isinstance(value, mx.nd.NDArray):
+                self.sw.add_histogram(tag=name, values=value, bins=100, global_step=checkpoint)
+            else:
+                self.sw.add_scalar(tag=name, value=value, global_step=checkpoint)
+        self.sw.flush()
```

Comment:
One minor comment, `flush()` would block the main thread and may affect the performance if logging heavily. Suggest not using it.

Reply:
good point. I will reconsider this if we start logging events more frequently. As it is right now, logging occurs only at checkpoints (e.g. every 4k updates). Overall loss in training speed should be negligible.